imageio/plugins/gdal.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2015, imageio contributors
+# imageio is distributed under the terms of the (new) BSD License.
+
+""" Plugin for reading gdal files.
+"""
+from __future__ import absolute_import, print_function, division
+
+from .. import formats
+from ..core import Format
+
+_gdal = None  # lazily loaded in load_lib()
+
+
+def load_lib():
+    global _gdal
+    try:
+        import osgeo.gdal as _gdal
+    except ImportError:
+        raise ImportError("The GDAL format relies on the GDAL package."
+                          "Please refer to http://www.gdal.org/"
+                          "for further instructions.")
+    return _gdal
+
+
+GDAL_FORMATS = ('.tiff', ' .tif', '.img', '.ecw', '.jpg', '.jpeg')
+
+
+class GdalFormat(Format):
+
+    """
+
+    Parameters for reading
+    ----------------------
+    None
+
+    """
+
+    def _can_read(self, request):
+        return request.filename.lower().endswith(GDAL_FORMATS)
```

Comment:
ECW is not a gdal format per se, (just like TIFF), but can have embedded geospatial data much like the others.  Since this is the only plugin providing that format, I say sure.